src/hotspot/share/gc/g1/g1CollectedHeap.hpp

```diff
@@ -1084,10 +1084,17 @@ class G1CollectedHeap : public CollectedHeap {
   inline HeapRegion* region_at(uint index) const;
   inline HeapRegion* region_at_or_null(uint index) const;
 
+private:
   // Return the next region (by index) that is part of the same
   // humongous object that hr is part of.
   inline HeapRegion* next_region_in_humongous(HeapRegion* hr) const;
 
+public:
+  // Iterate over the regions that the humongous object starting at the given
+  // region and apply the given method with the signature f(HeapRegion*) on them.
+  template <typename Functor>
```

Comment:
Maybe "Func" or "Function"? "Functor" means sth diff in some functional langs.

Reply:
Not sure we should care what it means in some functional langs if the meaning is clear for C++ lambdas.